File: src/main/java/com/qtech/bubbles/item/inventory/IInventory.java
package com.qtech.bubbles.item.inventory;

import com.qtech.bubbles.item.Item;

public interface IInventory {
    int size();

    Item[] getItems();

    void setItem(int index, Item item);

    void clear();

    Item getItem(int slot);

    void removeItem(int index);

    void tick();
}

File: src/main/kotlin/com/qtech/bubbles/item/inventory/IInventory.kt
package com.qtech.bubbles.item.inventory

import com.qtech.bubbles.item.ItemInstance

interface IInventory {
    fun size(): Int
    val itemInstances: Array<ItemInstance?>
    fun set(index: Int, itemInstance: ItemInstance?)
    fun clear()
    fun get(index: Int): ItemInstance?
    fun removeItem(index: Int)
    fun tick()
}